wgpu/examples/shadow/main.rs

```diff
@@ -213,12 +213,36 @@ impl framework::Example for Example {
         wgpu::Features::DEPTH_CLAMPING
     }
 
+    fn required_limits(adapter: &wgpu::Adapter) -> wgpu::Limits {
```

Comment:
I don't think this function should need an adapter.
According to the spec, you can ask for lower than baseline limits, and you'll still get the baseline. So this function should just ask for 0 storage buffers unconditionally. If there happen to be storage buffers supported, it can discover this by asking for device limits.

Reply:
So today, I think it should just check for `wasm` target. If it's wasm, request 0 storage buffers.

Reply:
According to the WGPU docs on the `Limits` struct:

> Once a device is requested, you may only use resources up to the limits requested even if the adapter supports “better” limits.

So that would mean that if we request a limit of 0, we can't use 1 storage buffer, even if the device supports it.

Does that mean we should just leave it how it is?

Reply:
It's a bit more nuanced. WebGPU spec requires implementations to return limits not worse than the baseline.
So if you request 0, it will still have to return the baseline limit.